### PR TITLE
Upgrade to Angular 14

### LINKED ISSUE #553: Cyclical dependency when installing @angular-devkit/build-angular
# Description

It is not possible to install all NPM package dependencies with `--strict-peer-deps`, which is an issue because the Renovatebot will use that to install all dependencies cleanly. There appears to be a cyclical dependency error when resolving the package `@angular-devkit/build-angular`.

# Steps to reproduce

1. Clone a clean copy of `mdm-ui`
2. Delete your local copy of the `package-lock.json` file
3. Delete the `node_modules` folder if it exists
4. Run `npm uninstall @angular/localize` and `npm uninstall @angular/language-service` - these are no needed, and uninstalling them will remove other resolution errors.
5. Now run `npm install --strict-peer-deps` to ensure all peer dependencies are valid.

# Expected behavior

`npm install ` should complete successfully without any errors.

# Additional context

Below is a copy of the resolution report that npm produces:
[eresolve-report.txt](https://github.com/MauroDataMapper/mdm-ui/files/8799214/eresolve-report.txt)

### LINKED ISSUE #554: Package dependencies preventing an upgrade to Angular 13 or higher
I was attempting an upgrade from Angular 12 to 13 for `mdm-ui` to try and resolve peer dependency issues. The upgrade process involves:

1. Removing the local `package-lock.json` file
2. Deleting all `node_modules` already on your local machine (to avoid conflicts)
3. Copy original list of dependencies from `package.json` _before_ making modifications.
4. Overwrite all package dependencies with latest versions of Angular packages
5. Run `npm install`
6. Then add back additional and 3rd party packages that the Angular upgrade process does not know about.

However, the process highlighted these package dependencies which will prevent an upgrade to Angular 13 because they do not support the latest version of Angular

1. `angular-user-idle`
2. `jodit-angular`

`angular-user-idle` is no longer supported. The `mdm-explorer` repo gets around this issue by directly importing the source code (with credit to the original authors) into the repo, maintaining it's own forked copy.

`jodit-angular` has this open issue for Angular 13 support - https://github.com/jodit/jodit-angular/issues/101 - but has not been completed (as of creating this issue).

Both of these are blockers for an Angular 13 upgrade.

### LINKED ISSUE #608: `mat-progress-spinner` component jumps in safari
In safari on a Mac, the `<mat-progress-spinner>` component jumps around and looks very unappealing.  This might be a styling issue, or maybe we need a new component.  Will try and upload a video to demonstrate if I can work out how.

### PR DESCRIPTION
**Pull request by jamesrwelch**
This is the final PR in the series - completing the upgrade to Angular 14.  As part of this PR we also upgrade Chart JS for the summary metadata charts (I couldn't resist a minor style tweak at the same time), and upgrade a number of other packages for consistency.

The PR for upgrading Jodit should ideally be accepted before this one, so leaving it in draft mode for now.

Fixes #608
Fixes #553
Fixes #554

### CODE REVIEW

**Comment by pjmonks** [comment]
Finding a significant issue with running the tests, almost half of them fail now because of importing `ngx-jodit`:

```
Cannot find module 'ngx-jodit' from 'src/app/shared/shared.module.ts'

    Require stack:
      src/app/shared/shared.module.ts
      src/app/testing/testing.module.ts
      src/app/testing/testing.helpers.ts
      src/app/bulk-edit/bulk-edit-profile.service.spec.ts

      63 | import { MauroItemTreeComponent } from './mauro-item-tree/mauro-item-tree.component';
      64 | import { ElementLabelComponent } from './element-label/element-label.component';
    > 65 | import { NgxJoditModule } from 'ngx-jodit';
         | ^
      66 |
      67 | @NgModule({
      68 |   declarations: [

      at Resolver._throwModNotFoundError (node_modules/jest-resolve/build/resolver.js:491:11)
      at Object.<anonymous> (src/app/shared/shared.module.ts:65:1)
```

The `ngx-jodit` module is correctly imported into the `SharedModule` and can be found by Angular at runtime, but Jest cannot find it. I've tried a number of things to see if I can resolve it, the closest I thought this was was related to this issue:

https://github.com/JsDaddy/ngx-mask/issues/962

`ngx-jodit` does not have a `main` field in its `package.json` file, so I've tried manually configuring Jest to map directly to those files, but without success. I'm at a bit of a loss on what to try next.

**Comment by pjmonks** [comment]
Issues with running the tests don't occur in the `develop` branch targeting Angular 12, so this test issue has somehow been introduced in the upgrade to Angular 14.

**Comment by jamesrwelch** [comment]
I've been battling this all afternoon... I was hoping you might know the answer!  
I have a few more ideas to try... I'm sure I had it working some point in the past

**Comment by jamesrwelch** [comment]
Reverted too much... not quite there...

**Comment by jamesrwelch** [comment]
@pjmonks I think this is now ready for review.

**Comment by pjmonks** [comment]
Found an issue with the Bulk Editor which seems to be new as part of this upgrade, but can be fixed separately. See issue #739

File: src/app/catalogue-search/catalogue-search-form/catalogue-search-form.component.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ import { MatFormFieldAppearance } from '@angular/material/form-field';
 export class CatalogueSearchFormComponent implements OnInit {
   @Input() appearance: MatFormFieldAppearance = 'outline';
   @Input() routeSearchTerm?: string = '';
-  formGroup: FormGroup = new FormGroup({});
+  formGroup: UntypedFormGroup = new UntypedFormGroup({});
```
Comment (pjmonks) [inline src/app/catalogue-search/catalogue-search-form/catalogue-search-form.component.ts:37]: I've heard about these new typed Angular Forms, would be interesting to refactor some of these at a later date.

**Review by pjmonks — CHANGES_REQUESTED** [review]
The branch now builds successfully. I've had a quick test clicking around the UI, only found one regression bug which I've created as a separate issue, but other than that cannot find any other issues.

I've requested one change to the config files to not break developer workflow, but once that's done I think I can approve it.

File: package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
     "dist-archive": "tar zcvf dist/mdm-ui-$npm_package_version$([ -z \"$MDM_UI_THEME_NAME\" ] && echo \"\" || echo \".$MDM_UI_THEME_NAME\").tgz -C dist mdm-ui-$npm_package_version$([ -z \"$MDM_UI_THEME_NAME\" ] && echo \"\" || echo \"-$MDM_UI_THEME_NAME\")",
     "test": "ng test",
     "test:watch": "ng test --watch",
-    "test-clearCache": "ng test --clearCache",
+    "test-clearCache": "ng test --clear-cache",
```
[the remark this answers is not in the document]
Comment (jamesrwelch) [reply package.json:12]: Fantastic catch!  I've pushed this change.

File: src/app/catalogue-search/catalogue-search-form/catalogue-search-form.component.ts (hunk 1 above)
Comment (jamesrwelch) [reply src/app/catalogue-search/catalogue-search-form/catalogue-search-form.component.ts:37]: Yes - the automatic migration didn't work for me so I had to figure this out myself, and decided Untyped versions would be best for now.  But I've added a ticket #742 for sorting it out later

**Comment by jamesrwelch** [comment]
Thanks again for your really thorough reviewing on this, @pjmonks !

**Review by pjmonks — APPROVED** [review]
I think this is in a great state to accept now, nice one! 🎉 I'll approve the PR but will give you the honour of merging 🏆

**Comment by julianpoemp** [comment]
Hi,
I'm the developer of [ngx-jodit](https://github.com/julianpoemp/ngx-jodit) and found out, that you were using ngx-jodit, but reverted back to jodit-angular. I just want to let you know that you can use Jodit v4 with ESM now and it supports template driven or reactive forms now (see [demo](https://github.julianpoemp.com/ngx-jodit/3.x/)) 🚀 . Feel free to test it :) . If there are any issues please create on on the [issues page.](https://github.com/julianpoemp/ngx-jodit/issues) 😄 .

**Pull request merged**